modules/dns/main.tf

```diff
@@ -2,6 +2,11 @@
 # Add DNS records
 ##############################################################################
 
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_inputs = [for record in var.dns_record_set : (record.type == "SRV" || record.type == "LOC" || record.type == "CAA") ? (record.data != null ? null : tobool("The dns_record_set data block cannot have null value for the following DNS record - ${record.name}")) : null]
```

Comment:
The error message `The dns_record_set data block` is not completely correct. It can be something like - `The data{} block in dns_record_set is not defined for the DNS record - test-example`

Reply:
Done